MSOE.MediaComplete/Settings.xaml.cs

```diff
@@ -100,6 +101,11 @@ private void BtnSelectHomeFolder_Click(object sender, RoutedEventArgs e)
 
             if (!_allDirs.Contains(SettingWrapper.HomeDir.FullPath))
             {
+                var withParent = _allDirs.Where(x => x.StartsWith(SettingWrapper.HomeDir.FullPath, StringComparison.Ordinal));
+                foreach (var parent in withParent)
+                {
+                    _allDirs.Remove(parent);
+                }
```

Comment:
_alldirs => we have a list of directories that the user has used previously as a root directory. 
if() => if the user selects a directory that already exists in the list, do not do the inner
.where() => returns all the directories the user has previously visited that start with the FULL PATH  of  the newly selected path, then removes them from the list containing all of the previously visited directories

BASICALLY, if you select the parent directory of a directory you've already visited, then we remove the child reference in our stored list